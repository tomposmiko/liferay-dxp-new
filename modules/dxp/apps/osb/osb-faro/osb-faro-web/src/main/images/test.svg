<svg fill="none" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
	<rect fill="#6B6C7E" height="20" rx="5" width="20" />
	<path d="M14.7653 13.6478C13.308 10.8283 11.8837 9.2493 11.8762 7.75V5.5C11.8762 5.5 12.1619 5.5007 12.2512 5.5C13.2184 5.4918 13.2937 4.0382 12.2512 4H7.75116C6.72389 4.05789 6.77545 5.46133 7.75116 5.5H8.12616V7.75C8.13975 9.25164 6.56006 11.0352 5.23702 13.6478C4.92225 14.2696 5.09452 15.8924 6.99202 16H13.0103C14.6774 15.9386 15.1429 14.3784 14.7653 13.6478ZM12.9285 14.5H7.05577C6.84577 14.5 6.57342 14.2649 6.74077 13.9075L7.37499 12.6252C9.27108 12.6252 9.97022 11.1231 11.5103 11.1241C11.5103 11.1241 13.1483 13.784 13.2058 13.8625C13.4507 14.1979 13.2137 14.5 12.9285 14.5Z" fill="white" />
</svg>