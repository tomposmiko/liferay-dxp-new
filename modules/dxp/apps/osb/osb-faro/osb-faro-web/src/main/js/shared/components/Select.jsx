import getCN from 'classnames';
import omitDefinedProps from 'shared/util/omitDefinedProps';
import React from 'react';
import {PropTypes} from 'prop-types';

class SelectItem extends React.Component {
	render() {
		const {children, className, ...otherProps} = this.props;

		return (
			<option className={className} {...otherProps}>
				{children}
			</option>
		);
	}
}

class Select extends React.Component {
	static defaultProps = {
		showBlankOption: false
	};

	static propTypes = {
		showBlankOption: PropTypes.bool
	};

	render() {
		const {
			children,
			className,
			showBlankOption,
			...otherProps
		} = this.props;

		return (
			<select
				className={getCN('form-control select-root', className)}
				{...omitDefinedProps(otherProps, Select.propTypes)}
			>
				{showBlankOption && <SelectItem />}

				{children}
			</select>
		);
	}
}

Select.Item = SelectItem;
export default Select;
