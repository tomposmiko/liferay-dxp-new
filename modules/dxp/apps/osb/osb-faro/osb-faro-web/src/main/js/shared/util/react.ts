import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
