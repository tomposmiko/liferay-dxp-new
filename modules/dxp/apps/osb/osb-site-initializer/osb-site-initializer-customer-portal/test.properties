##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Customer Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Customer Portal")\
        )