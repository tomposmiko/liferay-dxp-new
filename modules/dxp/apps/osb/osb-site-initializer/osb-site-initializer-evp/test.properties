##
## Test Batch
##

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP