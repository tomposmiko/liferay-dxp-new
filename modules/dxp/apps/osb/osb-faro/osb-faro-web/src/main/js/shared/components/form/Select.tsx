import getCN from 'classnames';
import HelpBlock from './HelpBlock';
import Label from './Label';
import React from 'react';
import Select from '../Select';
import {FieldProps} from 'formik';

interface IFormSelectProps
	extends FieldProps,
		React.HTMLAttributes<HTMLElement> {
	disabled: boolean;
	inline?: boolean;
	label: React.ReactNode;
	popover: {
		content: React.ReactNode;
		title: React.ReactNode;
	};
	required?: boolean;
	secondaryInfo?: React.ReactNode;
}

const FormSelect: React.FC<IFormSelectProps> = ({
	children,
	className,
	field,
	form,
	inline = false,
	label,
	onChange,
	popover,
	required = false,
	secondaryInfo,
	...otherProps
}) => {
	const {disabled} = otherProps;
	const {name} = field;

	const error = form.errors[name];
	const touched = form.touched[name];

	const classes = getCN(className, {
		'form-inline-group': inline,
		'has-error': touched && error,
		'has-success': touched && !error && !disabled
	});

	const handleChange = event => {
		field.onChange(event);

		if (onChange) {
			onChange(event);
		}
	};

	return (
		<div className={classes}>
			{label && (
				<Label htmlFor={name} popover={popover} required={required}>
					{label}
				</Label>
			)}

			{secondaryInfo && (
				<Label className='font-weight-normal' htmlFor={name}>
					<p>{secondaryInfo}</p>
				</Label>
			)}

			<Select
				{...otherProps}
				{...field}
				id={name}
				name={name}
				onChange={handleChange}
			>
				{children}
			</Select>

			<HelpBlock name={name} />
		</div>
	);
};

export default Object.assign(FormSelect, {Item: Select.Item});
