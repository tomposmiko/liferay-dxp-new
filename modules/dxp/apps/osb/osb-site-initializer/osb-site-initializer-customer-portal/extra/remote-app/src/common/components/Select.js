import ClayForm, {ClaySelect} from '@clayui/form';
import ClayIcon from '@clayui/icon';
import {useField} from 'formik';
import {required, validate} from '../utils/validations.form';
import WarningBadge from './WarningBadge';

const Select = ({
	groupStyle,
	helper,
	label,
	options,
	validations,
	...props
}) => {
	if (props.required) {
		validations = validations
			? [...validations, (value) => required(value)]
			: [(value) => required(value)];
	}

	const [field, meta] = useField({
		...props,
		validate: (value) => validate(validations, value),
	});

	const getStyleStatus = () => {
		if (meta.touched) {
			return meta.error ? ' has-error' : ' has-success';
		}

		return '';
	};

	return (
		<ClayForm.Group
			className={`w-100${getStyleStatus()} ${
				groupStyle ? groupStyle : ''
			}`}
		>
			<label>
				{`${label} `}

				{props.required && (
					<span className="inline-item-after reference-mark text-warning">
						<ClayIcon symbol="asterisk" />
					</span>
				)}

				<div className="position-relative">
					<ClayIcon className="select-icon" symbol="caret-bottom" />

					<ClaySelect {...field} {...props}>
						{options.map(({disabled, label, value}) => (
							<ClaySelect.Option
								disabled={disabled}
								key={value}
								label={label}
								value={value}
							/>
						))}
					</ClaySelect>
				</div>
			</label>

			{meta.touched && meta.error && props.required && (
				<WarningBadge>
					<span className="pl-1">{meta.error}</span>
				</WarningBadge>
			)}

			{helper && <div>{helper}</div>}
		</ClayForm.Group>
	);
};

export default Select;
