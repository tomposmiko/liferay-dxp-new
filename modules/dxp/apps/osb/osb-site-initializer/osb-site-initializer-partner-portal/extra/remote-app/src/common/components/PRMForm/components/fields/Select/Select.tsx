/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import {ClaySelectWithOption} from '@clayui/form';
import React from 'react';

import WrapperInput from '../common/components/WrapperInput';
import PRMFormFieldProps from '../common/interfaces/prmFormFieldProps';
import PRMFormFieldStateProps from '../common/interfaces/prmFormFieldStateProps';

interface IProps {
	emptyOptionMessage: string;
}

const Select = ({
	field,
	meta,
	label,
	emptyOptionMessage = 'No options available',
	options = [],
	required,
	...props
}: PRMFormFieldProps &
	React.ComponentProps<typeof ClaySelectWithOption> &
	PRMFormFieldStateProps<any> &
	IProps) => {
	const defaultOptions = {
		disabled: true,
		label: options.length ? 'Choose a option' : emptyOptionMessage,
		value: '',
	};

	const getValue = () => {
		if (typeof field.value === 'object') {
			if (field.value.id) {
				return String(field.value.id) || '';
			}

			return field.value.key || '';
		}

		return field.value || '';
	};

	return (
		<WrapperInput {...meta} label={label} required={required}>
			<ClaySelectWithOption
				options={[defaultOptions, ...options]}
				{...field}
				{...props}
				required={required}
				value={getValue()}
			/>
		</WrapperInput>
	);
};

export default Select;
