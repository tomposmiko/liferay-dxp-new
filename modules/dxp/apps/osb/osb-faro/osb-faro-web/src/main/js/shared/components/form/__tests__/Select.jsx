import React from 'react';
import Select from '../Select';
import {cleanup, render} from '@testing-library/react';
import {mockForm} from 'test/data';

jest.unmock('react-dom');

describe('Select', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(
			<Select field={{name: 'test'}} form={mockForm()}>
				<Select.Item value='red'>{'red'}</Select.Item>
			</Select>
		);
		expect(container).toMatchSnapshot();
	});

	it('renders w/ label', () => {
		const {container} = render(
			<Select field={{name: 'test'}} form={mockForm()} label='foo' />
		);
		expect(container.querySelector('label')).not.toBeNull();
	});

	it('renders w/ error', () => {
		const {container} = render(
			<Select
				field={{name: 'test'}}
				form={mockForm({
					errors: {test: 'can not be test'},
					touched: {test: true}
				})}
			>
				<Select.Item value='red'>{'red'}</Select.Item>
			</Select>
		);

		expect(container.querySelector('.has-error')).not.toBeNull();
	});
});
