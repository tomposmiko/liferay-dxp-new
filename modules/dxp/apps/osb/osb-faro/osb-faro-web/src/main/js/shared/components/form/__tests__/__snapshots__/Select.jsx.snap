// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
<div>
  <div
    class=""
  >
    <select
      class="form-control select-root"
      id="test"
      name="test"
    >
      <option
        value="red"
      >
        red
      </option>
    </select>
  </div>
</div>
`;
